src/Kriti/Parser/Lexer.x

```diff
@@ -0,0 +1,133 @@
+-- This also means that '{' will be in its own `TokStringLit` required
+-- a post lexing pass to concatenate adjacent `TokStringLit` tokens.
+
+-- In <expr> mode we allow the subset of <0> mode rules required to
+-- construct our allowed expressions: Bools, Numbers, and Paths.
+
+-- Exit <expr> mode if we match '}}'
+<expr> \} \} { \b -> (popStartCode *> symbol SymDoubleCurlyClose b) }
+
+-- Exit <string> Mode if we match '\"'
+<string> \" { \b -> (popStartCode *> symbol SymStringEnd b) }
+
+-- String literals can be constructed with single quotes. These are
+-- only used for object lookup and are thus only available in <expr>
+-- mode.
+<expr> \'                              { \b -> pushStartCode literal *> symbol SymSingleQuote b }
+<literal> (\\ \\ | \\ \` | [^ \' \{ ])+ { token TokStringLit }
+<literal> \'                            { \b -> (popStartCode *> symbol SymSingleQuote b) }
+
+<0, expr> \-? $digit+                                       { token (\loc -> TokIntLit (unLoc loc) (read . T.unpack <$> loc)) }
+<0, expr> \-?(0|[1-9][0-9]*)(\.[0-9]+)?([eE][\+\-]?[0-9]+)? { token (\loc -> TokNumLit (unLoc loc) (read . T.unpack <$> loc)) }
+<0> \:                                                      { symbol SymColon }
+<0, expr> \.                                                { symbol SymDot }
+<0, expr> \,                                                      { symbol SymComma }
+<0, expr> \==                                                     { symbol SymEq }
+<0, expr> \>                                                      { symbol SymGt }
+<0, expr> \<                                                      { symbol SymLt }
+<0, expr> \<                                                      { symbol SymLt }
+<0, expr> \&\&                                                    { symbol SymAnd }
+<0, expr> \|\|                                                    { symbol SymOr }
+<0, expr> \_                                                      { symbol SymUnderscore }
+<0, expr> \:\=                                                    { symbol SymAssignment }
+<0, expr> \{                                                      { symbol SymCurlyOpen }
+<0, expr> \}                                                      { symbol SymCurlyClose }
+
+-- When we encounter a `{{` we push an <expr> code. This way we know
+-- when to parse two `}` as `SymDoubleCurlyClose` rather then as two
+-- `SymCurlyClose`
+<0, expr> \{ \{                                             { \b -> (pushStartCode expr *> symbol SymDoubleCurlyOpen b) }
+
+<0, expr> \[                                                { symbol SymSquareOpen }
+<0, expr> \]                                                { symbol SymSquareClose }
+<0, expr> \(                                                { symbol SymParenOpen }
+<0, expr> \)                                                { symbol SymParenClose }
+
+{
+-- | Our monadic wrapper for `alexScan`. We hold the latest
+scan :: Parser Token
```

Comment:
This is our `Parser Token` monadic wrapper for the [alexScan](https://github.com/simonmar/alex/blob/ab87af1803a5e2f2c09b09eb024dc6ec9f44b0e3/src/Scan.hs#L297) function.

Reply:
This comment looks incomplete; it ends with `We hold the latest`, but it seems like there should be more commentary here.